content/posts/macos/macOS Security Overview/index.md

```diff
@@ -0,0 +1,347 @@
+---
+title: "macOS Security Overview"
+date: 2024-08-09
+tags: ['macOS', 'Security']
+author: fria, Tommy
+---
+
+macOS comes equipped with many security features to keep you safe. Check out the [Apple Platform Security](https://support.apple.com/guide/security/welcome/web) page for more detail about the security features in a Mac. This post assumes you're running on official Apple hardware that's officially supported by the latest stable version of macOS.
+
+## Encryption
+
+By default, your macOS install is encrypted, but it will automatically unlock on boot. Turning on [FileVault](https://support.apple.com/guide/mac-help/protect-data-on-your-mac-with-filevault-mh11785/mac) will require a user password to unlock the volume. It also makes it so that you need to enter a user password to enter recovery mode.
+
+All encryption keys are handled by the Secure Enclave. Swap space is also [encrypted](https://support.apple.com/en-euro/guide/mac-help/mh11852/mac).
+
+Your Mac is at its most secure when it's fully off and the data is at rest. Depending on your threat model, it might behoove you to turn your Mac off completely whenever you're not using it, especially since Macs do not have memory encryption.
```

Comment:
macOS can be forced to clear FileVault keys from memory when hibernating by executing:
```bash
sudo pmset -a hibernatemode 25
```
and installing [`DestroyFVKeyOnStandby` profile](https://github.com/usnistgov/macos_security/blob/main/rules/os/os_hibernate_mode_destroyfvkeyonstandby_enable.yaml).

Reply:
Is this actually documented anywhere by apple or nah?

Reply:
`man pmset`, online version e.g. at https://www.unix.com/man_page/osx/1/pmset/

> hibernatemode = 25 (binary 0001 1001) is only settable via pmset. The system will store a copy of memory to persistent storage (the disk), and will remove power to memory. The system will restore from disk image. If you want "hibernation" - slower sleeps, slower wakes, and better battery life, you should use this setting.

Configuration profiles documentation: https://developer.apple.com/documentation/devicemanagement/fdefilevaultoptions.

All configuration profiles in an official Git repository; check new commits after every system update to see the difference between releases: https://github.com/apple/device-management/tree/release/mdm/profiles.